lib/protobuf/protoc/generator/enum.ex

```diff
@@ -10,8 +10,18 @@ defmodule Protobuf.Protoc.Generator.Enum do
     fields = Enum.map(desc.value, fn f -> generate_field(f) end)
     msg_name = Util.mod_name(ctx, ns ++ [name])
     generate_desc = if ctx.gen_descriptors?, do: desc, else: nil
+    type = generate_type(desc.value)
 
-    Protobuf.Protoc.Template.enum(msg_name, msg_opts(ctx, desc), fields, generate_desc)
+    Protobuf.Protoc.Template.enum(msg_name, msg_opts(ctx, desc), fields, type, generate_desc)
+  end
+
+  def generate_type(fields) do
+    field_values =
+      fields
+      |> Enum.map(fn f -> ":#{f.name} | #{f.number}" end)
```

Comment:
I think we only need `integer` instead of every number because the reason we may get integer as enum is the protobuf may include an unknown enum, which should also be handled based on Protobuf's spec.

Reply:
Sure sounds reasonable :)